Source/Xamarin/Prism.DryIoc.Forms/DryIocExtensions.cs

```diff
@@ -0,0 +1,43 @@
+using DryIoc;
+using Prism.Common;
+using Xamarin.Forms;
+
+namespace Prism.DryIoc
+{
+    /// <summary>
+    /// Extension methods to register a <see cref="Xamarin.Forms.Page"/> with <see cref="PageNavigationRegistry"/> using <see cref="IContainer"/>
+    /// </summary>
+    public static class DryIocExtensions
+    {
+        /// <summary>
+        /// Registers a Page for navigation using a convention based approach, which uses the name of the Type being passed in as the unique name.
+        /// </summary>
+        /// <typeparam name="TPage">The Type of Page to register</typeparam>
+        public static void RegisterTypeForNavigation<TPage>(this IContainer container) where TPage : Page
+        {
+            container.RegisterTypeForNavigation<TPage>(typeof(TPage).FullName);
```

Comment:
@joacar you can look at the updated Unity version which was updated for more clarity. It's not about Pages and classes. Since Prism is about MVVM it's really about Views and ViewModels which with Xamarin Forms are fulfilled with Pages and BindableBase types. You navigate to View's not ViewModel's which is why for TView we just get the type Name, and to discourage the bad practice of navigating to a ViewModel we're taking the FullName of the TViewModel.